gefserver/pier/internal/dckr/dckr.go

```diff
@@ -368,13 +370,18 @@ func (c Client) StartImage(id string, repoTag string, cmdArgs []string, binds []
 		MemorySwap: limits.MemorySwap,
 	}
 
+	createContainerContext, cancel := context.WithTimeout(context.Background(), time.Duration(preparationTimeOut)*time.Second)
+	defer cancel()
```

Comment:
https://golang.org/pkg/context/
createContainerContext is a context for creating a container, context with a timeout, defer cancel()  releases resources if slowOperation completes before timeout elapses